price-feeder/config/config.go

```diff
@@ -117,32 +117,32 @@ type (
 	// Telemetry defines the configuration options for application telemetry.
 	Telemetry struct {
 		// Prefixed with keys to separate services
-		ServiceName string `toml:"service_name"`
+		ServiceName string `toml:"service_name" mapstructure:"service-name"`
 
 		// Enabled enables the application telemetry functionality. When enabled,
 		// an in-memory sink is also enabled by default. Operators may also enabled
 		// other sinks such as Prometheus.
-		Enabled bool `toml:"enabled"`
+		Enabled bool `toml:"enabled" mapstructure:"enabled"`
 
 		// Enable prefixing gauge values with hostname
-		EnableHostname bool `toml:"enable_hostname"`
+		EnableHostname bool `toml:"enable_hostname" mapstructure:"enable-hostname"`
 
 		// Enable adding hostname to labels
-		EnableHostnameLabel bool `toml:"enable_hostname_label"`
+		EnableHostnameLabel bool `toml:"enable_hostname_label" mapstructure:"enable-hostname-label"`
 
 		// Enable adding service to labels
-		EnableServiceLabel bool `toml:"enable_service_label"`
+		EnableServiceLabel bool `toml:"enable_service_label" mapstructure:"enable-service-label"`
 
 		// GlobalLabels defines a global set of name/value label tuples applied to all
 		// metrics emitted using the wrapper functions defined in telemetry package.
 		//
 		// Example:
 		// [["chain_id", "cosmoshub-1"]]
-		GlobalLabels [][]string `toml:"global_labels"`
+		GlobalLabels [][]string `toml:"global_labels" mapstructure:"global-labels"`
 
-		// Type determines which type of telemetry to use
-		// Valid values are "prometheus" or "generic"
-		Type string `toml:"type"`
+		// PrometheusRetentionTime, when positive, enables a Prometheus metrics sink.
+		// It defines the retention duration in seconds.
+		PrometheusRetentionTime int64 `toml:"prometheus_retention" mapstructure:"prometheus-retention-time"`
 	}
```

Comment:
Isn't it needed for decoding the toml?

Reply:
Chains are using toml, it already works in the chain. I'm not sure if the decoding goes through the mapstructure. We can test what will be the right name in toml (`prometheus-retention-time` or `prometheus_retention_time`).

Reply:
It looks like the SDK parses these line-by-line: 
https://github.com/cosmos/cosmos-sdk/blob/23576d3d49de3be415bf8c55fbf4c67ea690875b/server/config/config.go#L314
Along with custom logic for the global labels:
https://github.com/cosmos/cosmos-sdk/blob/23576d3d49de3be415bf8c55fbf4c67ea690875b/server/config/config.go#L291

Is it cleaner to do it this way or just leave the structure in?

Reply:
`Viper` object it's used - it will use prioritized setup based on env variables > config file.
I think we should do the same to enable configuration via env variables.